### PR TITLE
Add bbduk and prinseq complexity (entropy or dust-based) short read filtering

### LINKED ISSUE #8: Preprocessing: short read complexity filtering
### Description of feature

We should add a step that allows removal of low complexity reads prior profiling, to reduce runtime and unspecific hits

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

Completely closes #8 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

---

- Adds BBduk  based complexity filter to a complexity filter subworkflow
- Adds PRINSEQ++ based complxiter filter to a complexity filter subworkflow
- Removes the CAT_FASTQ module for the time being (until we add run merging again)
- Syncronises subworkflow header formatting
- Fixes MultiQC channel insertion for all tools that were missing by moving them all to the end of the main workflow, rather than within subworkflows

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -155,6 +155,19 @@ process {
         ]
     }
 
+    withName: PRINSEQPLUSPLUS {
+        ext.args =  [
+                params.shortread_complexityfilter_prinseqplusplus_mode == 'dust' ? "-lc_dust=${params.shortread_complexityfilter_prinseqplusplus_dustscore}" : "-lc_entropy=${params.shortread_complexityfilter_entropy}",
+                "-trim_qual_left=0 -trim_qual_left=0 -trim_qual_window=0 -trim_qual_step=0"
```
Comment (Midnighter) [inline conf/modules.config:161]: I don't know the program so I can't really judge these options. I'm guessing you're configuring dust and disabling quality-based trimming.

File: nextflow_schema.json [hunk 2]
```diff
@@ -321,6 +330,18 @@
...
         "shortread_complexityfilter": {
             "type": "boolean"
+        },
+        "shortread_complexityfilter_entropy": {
+            "type": "number",
+            "default": 0.3
+        },
+        "shortread_complexityfilter_prinseqplusplus_mode": {
+            "type": "string",
+            "default": "entropy"
```
Comment (Midnighter) [inline nextflow_schema.json:331]: Should this be an enum, too, with a fixed set of options?

**Review by Midnighter — APPROVED** [review]
Some very minor comments. Otherwise it looks like a great addition.

File: conf/modules.config (hunk 1 above)
Comment (jfy133) [reply conf/modules.config:161]: Me neither.


When I look at the help message those are the only values with defaults, but we don't want to do so any read trimming here, so I'm deactivating by setting to 0 for safety

So here I'm saying if user selects dust, use the dust argument, and if not the entropy argument. In both cases we append to it the disabling of trimming

File: nextflow_schema.json (hunk 2 above)
Comment (jfy133) [reply nextflow_schema.json:331]: Uups yes. Will correct in the morning 👍

**Review by Midnighter — APPROVED** [review]
Nice, more pressure for GH Actions 😁

**Comment by jfy133** [comment]
:fire:

**Comment by jfy133** [comment]
So found out bbduk is particular picky in some cases, so going to have to create an additional module to allow it to run :roll_eyes:

**Comment by Midnighter** [comment]
I think you need to merge `dev` again or rebase so the AdapterRemoval refactor is in.

**Comment by jfy133** [comment]
> ~I think you need to merge `dev` again or rebase so the AdapterRemoval refactor is in.~
> 
> I was looking at an outdated version.

Yeah I fucked up the merging when pulling your stuff in

**Pull request merged**